src/ZODB/tests/testMVCCMappingStorage.py

```diff
@@ -33,6 +33,15 @@
     )
 
 class MVCCTests:
+    def checkClosingNestedDatabasesWorks(self):
+        # This tests for the error described in
+        # https://github.com/zopefoundation/ZODB/issues/45
+        db1 = DB(self._storage)
+        db2 = DB(None, databases=db1.databases, database_name='2')
+        db1.open().get_connection('2')
+        db1.close()
+        db2.close()
```

Comment:
Zope stores all dbs that have been created. On shutdown, it closes each of them, I am mirroring this here.